pytorch_lightning/core/lightning.py

```diff
@@ -362,6 +370,27 @@ def log(
             # reset any tensors for the new hook name
             results.reset(metrics=False, fx=self._current_fx_name)
 
+        if metric_attribute is None and isinstance(value, Metric):
+            if self._metric_attributes is None:
+                # compute once
+                self._metric_attributes = {
+                    id(module): name
+                    for name, module in self.named_children() if isinstance(module, Metric)
+                }
```

Comment:
How does this work if someone uses `ModuleDict` to store their `Metric`s in the LightningModule?

Reply:
Ah, I see this was addressed in https://github.com/PyTorchLightning/pytorch-lightning/pull/8181! Great :)